### PR TITLE
Feature/602 sri zkp component retrieve content addressable hash document

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

CRUD management of Content Addressable Hash. This PR retrieves the document from the CCSMAnchorHash stored in the local storage.

## Related Issue

#602 : Generate Content Addressable Hash

[Related document](https://docs.google.com/document/d/1mkO5oou0TWk6kVHbTB01dyj7wuNGTfR9w6PAxSgXdGI/edit?usp=sharing)

## Motivation and Context

Retrieves the document related to the CCSMAnchorHash.

## How Has This Been Tested

- Unit tests have been written for the API

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Review by skosito — COMMENTED** [review]
@biscuitdey it seems that this PR contains changes of other PR as well
I don't want to mess with it to not accidentally remove some change, but I think it should be just switching base branch to point to `feature/602-sri-zkp-component-generate-content-addressable-hash` instead of `main` and the diff will be showed correctly, like this it is a bit hard to know which changes are covered with which PR

File: examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHash.agent.ts [hunk 1]
```diff
@@ -36,13 +38,11 @@ export class CCSMAnchorHashAgent {
...
   ): string {
-    const hash = this.convertDocumentToHash(inputForProofVerification);
-    return hash;
+    return this.convertTextToHash(inputForProofVerification);
   }
 
-  public convertDocumentToHash(document: string) {
-    //TODO: Convert document into payload using merkleTree service
-    const hash = document;
+  public convertTextToHash(text: string) {
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHash.agent.ts:44]: may be this piece can be reused, can we extract this out as util?

File: examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHashLocalStorage.agent.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    if (!ccsmAnchorHashModel) {
+      throw new NotFoundException(CCSM_ANCHOR_HASH_NOT_FOUND_ERR_MESSAGE);
+    }
+
+    const documentModel = ccsmAnchorHashModel.document;
+    return this.mapper.map(documentModel, Document, Document);
+  }
+
+  async createNewCCSMAnchorHash(
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHashLocalStorage.agent.ts:40]: do we really need the "New" text here?

File: examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHashLocalStorage.agent.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    });
+
+    return this.mapper.map(
+      newCCSMAnchorHashModel,
+      CCSMAnchorHash,
+      CCSMAnchorHash,
+    );
+  }
+
+  async createNewDocument(text: string): Promise<Document> {
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHashLocalStorage.agent.ts:71]: do we really need the "New" text here?

File: examples/bri-3/src/bri/zeroKnowledgeProof/capabilities/createCCSMAnchorHash/createCCSMAnchorHashCommand.handler.ts [hunk 4]
```diff
@@ -9,18 +10,25 @@ export class CreateCCSMAnchorHashCommandHandler
...
       command.document,
     );
 
-    await this.storageAgent.storeAnchorHashOnCCSM(newCCSMAnchorHash);
+    const newCCSMAnchorHash = await this.agent.createNewCCSMAnchorHash(
+      command.ownerAccount.id,
+      newDocument,
+    );
+
+    await this.localStorageAgent.createNewCCSMAnchorHash(newCCSMAnchorHash);
```
Comment (Manik-Jain) [inline examples/bri-3/src/bri/zeroKnowledgeProof/capabilities/createCCSMAnchorHash/createCCSMAnchorHashCommand.handler.ts:29]: why do we need the localStorage?

**Comment by ognjenkurtic** [comment]
> @biscuitdey it seems that this PR contains changes of other PR as well I don't want to mess with it to not accidentally remove some change, but I think it should be just switching base branch to point to `feature/602-sri-zkp-component-generate-content-addressable-hash` instead of `main` and the diff will be showed correctly, like this it is a bit hard to know which changes are covered with which PR

Yes, please either do this or we wait for the other PR to be merged into main before we continue with reviewing

**Comment by ognjenkurtic** [comment]
> > @biscuitdey it seems that this PR contains changes of other PR as well I don't want to mess with it to not accidentally remove some change, but I think it should be just switching base branch to point to `feature/602-sri-zkp-component-generate-content-addressable-hash` instead of `main` and the diff will be showed correctly, like this it is a bit hard to know which changes are covered with which PR
> 
> Yes, please either do this or we wait for the other PR to be merged into main before we continue with reviewing

@biscuitdey

**Comment by biscuitdey** [comment]
> > > @biscuitdey it seems that this PR contains changes of other PR as well I don't want to mess with it to not accidentally remove some change, but I think it should be just switching base branch to point to `feature/602-sri-zkp-component-generate-content-addressable-hash` instead of `main` and the diff will be showed correctly, like this it is a bit hard to know which changes are covered with which PR
> > 
> > 
> > Yes, please either do this or we wait for the other PR to be merged into main before we continue with reviewing
> 
> @biscuitdey

I'll just wait for the other PR to merge first and then update this.

**Comment by biscuitdey** [comment]
Will revisit this PR after State Object has been implemented

**Pull request closed without merging**